### PR TITLE
Task sf/j22 add handleerror middleware

### PR DESCRIPTION
**Pull request by Sayonarko**

### CODE REVIEW

**Comment by Sayonarko** [comment]
[Create `handleError` middleware for redux](https://trello.com/c/LCyAGqxd/22-create-handleerror-middleware-for-redux)

File: frontend/src/middlewares/middlewares.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from './handle-error/handle-error.middleware';
```
Comment (ghost) [inline frontend/src/middlewares/middlewares.ts]:
```suggestion
export { handleError } from './handle-error/handle-error.middleware';
```

File: frontend/src/middlewares/handle-error/handle-error.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { notification as notificationService } from 'services/services';
+
+const handleError: Middleware = () => (next) => (action): void => {
+  const hasErrorMessage = Boolean(action.payload?.error?.message);
+  if (hasErrorMessage) {
+    notificationService.error('Error', action.payload.error.message);
+  }
+  next(action);
```
Comment (ghost) [inline frontend/src/middlewares/handle-error/handle-error.middleware.ts]:
```suggestion
  const hasErrorMessage = Boolean(action?.error?.message);
  
  if (hasErrorMessage) {
    notificationService.error('Error', action.payload.error.message);
  }
  
  next(action);
```
awesome!

**Review by ghost — CHANGES_REQUESTED** [review]
Very good!

Also, change the check status in the http service a bit:

```
  private async checkStatus(response: Response): Promise<Response> {
    if (!response.ok) {
      const parsedException = await response.json().catch(() => ({
        message: response.statusText,
      }));

      throw new HttpError({
        status: response.status,
        message: parsedException?.message,
      });
    }
```

**Review by ghost — APPROVED** [review]
LGTM 👍

**Review by liza-veis — APPROVED** [review]

**Review by yevhenprybytok — APPROVED** [review]

**Pull request merged**